server/text_generation_server/utils/adapter.py

```diff
@@ -194,3 +210,70 @@ def load_module_map(
         adapter_weights, weight_names
     )
     return module_map, adapter_config, adapter_weight_names, adapter_tokenizer
+
+
```

Comment:
Isn't this super brittle? It only works for llama, no?

Reply:
its a bit brittle but should work in most cases, currently this will correctly load in weight for llama, mistral and gemma type models (just added some updates and tests too). The intention of moving this code here is to avoid mapping the weights inside of each models code.. currently this is the best approach I have, however I'm happy to make any changes!

Reply:
I guess we could have a mapping table here per model type? But probably better for another PR.